client/dashboard/plugins/manage/fields/name.tsx

```diff
@@ -1,19 +1,52 @@
+import Gridicon from '@automattic/components/src/gridicon';
 import { Link } from '@tanstack/react-router';
+import { __experimentalHStack as HStack } from '@wordpress/components';
 import { __ } from '@wordpress/i18n';
+import clsx from 'clsx';
 import { pluginRoute } from '../../../app/router/plugins';
 import type { PluginListRow } from '../types';
 import type { Field } from '@wordpress/dataviews';
 
+const ICON_SIZE = 32;
+const FALLBACK_ICON_SIZE = 24;
+
 export const nameField: Field< PluginListRow > = {
 	id: 'name',
 	label: __( 'Plugin' ),
 	enableHiding: false,
 	enableSorting: true,
 	getValue: ( { item } ) => item.name,
-	render: ( { item, field } ) => (
-		<Link to={ pluginRoute.to } params={ { pluginId: item.slug } }>
-			{ field.getValue( { item } ) }
-		</Link>
-	),
+	render: ( { item, field } ) => {
+		let icon;
+
+		if ( typeof item.icons === 'string' ) {
+			icon = <img src={ item.icons } alt={ item.name } width={ ICON_SIZE } height={ ICON_SIZE } />;
+		} else if ( item.icons?.[ '1x' ] ) {
+			icon = (
+				<img
+					src={ item.icons[ '1x' ] }
+					alt={ item.name }
+					width={ ICON_SIZE }
+					height={ ICON_SIZE }
+				/>
+			);
+		} else {
+			icon = (
+				<Gridicon icon="plugins" size={ FALLBACK_ICON_SIZE } className="plugin-icon-fallback" />
+			);
+		}
```

Comment:
We could simplify the code a bit, maybe something like this should work:
```suggestion
const src =
	typeof item.icons === "string"
		? item.icons
		: item.icons?.["1x"];

const icon = src ? (
	<img src={src} alt={item.name} width={ICON_SIZE} height={ICON_SIZE} />
) : (
	<Gridicon
		icon="plugins"
		size={FALLBACK_ICON_SIZE}
		className="plugin-icon-fallback"
	/>
);
```

Reply:
Nice, thanks for the suggestion.